fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/SplitHelper.java

```diff
@@ -916,6 +950,38 @@ private boolean appendNext(@Nonnull KeyValue kv) {
                             .addLogInfo(LogMessageKeys.FOUND_INDEX, index);
                 }
             }
+            logNextKey(done);
+            return done;
+        }
+
+        private void logFirstKey(boolean done) {
+            logKey("found first key in new split record", done);
+        }
+
+        private void logNextKey(boolean done) {
+            logKey("found next key in split record", done);
+        }
+
+        private void logEndFound() {
+            logKey("end key found for split record", true);
+        }
+
+        private void logKey(@Nonnull String staticMessage, boolean done) {
+            if (LOGGER.isTraceEnabled()) {
+                KeyValueLogMessage msg = KeyValueLogMessage.build(staticMessage,
+                        LogMessageKeys.KEY_TUPLE, nextKey,
+                        LogMessageKeys.SPLIT_REVERSE, reverse,
+                        LogMessageKeys.SPLIT_NEXT_INDEX, nextIndex,
+                        LogMessageKeys.KNOWN_LAST_KEY, done,
+                        LogMessageKeys.SUBSPACE, ByteArrayUtil2.loggable(subspace.getKey()));
+                sizeInfo.addSizeLogInfo(msg);
+                long currentNanos = System.nanoTime();
+                if (readLastKeyNanos != 0) {
+                    msg.addKeyAndValue(LogMessageKeys.READ_LAST_KEY_MICROS, TimeUnit.NANOSECONDS.toMicros(currentNanos - readLastKeyNanos));
+                }
+                readLastKeyNanos = currentNanos;
```

Comment:
No biggies, but I guess it could be `readLastKeyNanos = System.nanoTime()` if you wanted to factor out the update to the log msg. And perhaps could init the `readLastKeyNanos = System.nanoTime()` right before you start the unspilt cursor to capture the time it takes to get the first key (could also get rid of the _if (readLastKeyNanos != 0)_. But this will capture timing info as is.

Reply:
Right. That was intentional to keep track of total time since the last time a key was read (so, like, if it turns out that adding a key to the map is the bottleneck (which I doubt, but, like, who knows), then we can have that time captured).